### PR TITLE
Make minArticlesBeforeShowingBanner configurable at the test level

### PR DESCRIPTION
**Pull request by jranks123**
## What does this change?
We would like to be able to set `minArticlesBeforeShowingBanner` configurable at the test level, so that we can run tests which change this parameter.

Also, this PR creates a new type, `EngagementBannerTestParams`, used when creating Engagement Banner test variants,  which contains all the values of `EngagementBannerParams` that a test may wish to configure, set as optional. This reduces boilerplate in the test files because, for example, we may want to specify only one parameter change in a variant, so by having this extra type with all the configurable parameters optional, we don't have to recreate the whole `EngagementBannerParams` every time we write a new variant.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/ui/first-pv-consent-plus-engagement-banner.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
     canShow as canShowEngagementBanner,
     hideBanner as hideEngagementBanner,
     deriveBannerParams as deriveEngagementBannerParams,
-    getBannerHtml as getEngagementBannerHtml,
+    bannerParamsToHtml as getEngagementBannerHtml,
```
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/ui/first-pv-consent-plus-engagement-banner.js]: `getEngagementBannerHtml` -> `engagementBannerParamsToHtml`

**Review by johnduffell — APPROVED** [review]
great thanks for updating all that!

**Comment by johnduffell** [comment]
looks like a few tests need updating and conflicts unfortunately!

**Review by tomrf1 — APPROVED** [review]
nice. No requirement to configure this from the google doc?

**Pull request merged**